File: 50. Pow(x, n).py
class Solution(object):
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        if n >= 0:
            try:
                return self.power(x, n)
            except OverflowError:
                return float(0)
        else:
            n = -n
            try:
                return float(1 / self.power(x, n))
            except OverflowError:
                return float(0)
        
    def power(self, x, n):
        if n == 0:
            return float(1)
        elif n == 1:
            return float(x)
        return float(self.power(x, n/2)**2 * self.power(x, n%2))

File: 50. Pow(x, n).cpp
class Solution {  
public:  
    double myPow(double x, int n) {  
        double temp = 0;
        if(n < 0)  
            temp = 1.0 / power(x, -n);  
        else  
            temp = power(x, n);
        if (!isfinite(temp))
            return 0.0;
        return temp;
    }  
  
    double power(double x, int n) {  
        if(n == 0)  
            return 1.0;
        double temp=power(x, n/2);
        
        if(n % 2 == 0)  
            return temp * temp;  
        else  
            return temp* temp * x;  
    }  
}; 
